src/orion/core/worker/producer.py

```diff
@@ -175,7 +188,10 @@ def _update_algorithm(self, completed_trials):
 
         new_completed_trials = []
         for trial in completed_trials:
-            if trial not in self.trials_history:
+            # if trial not in self.trials_history:
```

Comment:
```suggestion

```

And do we still need this `trials_history` for producer since we now have this `has_observed`?

Reply:
Yes, we do not use for this anymore but it's still useful to set parent trials (for reproducibility) in producer.register_trial().

Reply:
Yes, we do not use for this anymore but it's still useful to set parent trials (for reproducibility) in producer.register_trial().